test/unit/colorspace_spec.js

```diff
@@ -0,0 +1,563 @@
+
+        expect(ColorSpace.isDefaultDecode([0, 1, 0, 1, 0, 1, 0, 1], 4))
+          .toBeTruthy();
+        expect(ColorSpace.isDefaultDecode([1, 0, 0, 1, 0, 1, 0, 1], 4))
+          .toBeFalsy();
+      });
+  });
+
+  describe('DeviceGrayCS', function () {
+    it('should handle the case when cs is a Name object', function () {
+      let cs = new Name('DeviceGray');
+      let xref = new XRefMock([{
+        ref: new Ref(10, 0),
+        data: new Dict(),
+      }]);
+      let res = new Dict();
+
+      let colorSpace = ColorSpace.parse(cs, xref, res);
+
+      let testSrc = new Uint8Array([27, 125, 250, 131]);
+      let testDest = new Uint8Array(4 * 4 * 3);
+      let expectedDest = new Uint8Array(
+        [27, 27, 27,
+          27, 27, 27,
+          125, 125, 125,
+          125, 125, 125,
+          27, 27, 27,
+          27, 27, 27,
+          125, 125, 125,
+          125, 125, 125,
+          250, 250, 250,
+          250, 250, 250,
+          131, 131, 131,
+          131, 131, 131,
+          250, 250, 250,
+          250, 250, 250,
+          131, 131, 131,
+          131, 131, 131
+        ]
+      );
+      colorSpace.fillRgb(testDest, 2, 2, 4, 4, 4, 8, testSrc, 0);
+
+      expect(colorSpace.getRgb(new Float32Array([0.1]), 0)).toEqual(
+        new Uint8Array([
+          25, 25, 25
+        ]));
+      expect(colorSpace.getOutputLength(2, 0)).toEqual(6);
+      expect(colorSpace.isPassthrough(8)).toBeFalsy();
+      expect(testDest).toEqual(expectedDest);
+    });
+    it('should handle the case when cs is an indirect object', function () {
+      let cs = new Ref(10, 0);
+      let xref = new XRefMock([{
+        ref: cs,
+        data: new Name('DeviceGray'),
+      }]);
+      let res = new Dict();
+
+      let colorSpace = ColorSpace.parse(cs, xref, res);
+
+      let testSrc = new Uint8Array([27, 125, 250, 131]);
+      let testDest = new Uint8Array(3 * 3 * 3);
+      let expectedDest = new Uint8Array(
+        [27, 27, 27,
+          27, 27, 27,
+          125, 125, 125,
+          27, 27, 27,
+          27, 27, 27,
+          125, 125, 125,
+          250, 250, 250,
+          250, 250, 250,
+          131, 131, 131
+        ]
+      );
+      colorSpace.fillRgb(testDest, 2, 2, 3, 3, 3, 8, testSrc, 0);
+
+      expect(colorSpace.getRgb(new Float32Array([0.2]), 0)).toEqual(
+        new Uint8Array(
+          [51, 51, 51]));
+      expect(colorSpace.getOutputLength(3, 1)).toEqual(12);
+      expect(colorSpace.isPassthrough(8)).toBeFalsy();
+      expect(testDest).toEqual(expectedDest);
+    });
+  });
+
+  describe('DeviceRgbCS', function () {
+    it('should handle the case when cs is a Name object', function () {
+      let cs = new Name('DeviceRGB');
+      let xref = new XRefMock([{
+        ref: new Ref(10, 0),
+        data: new Dict(),
+      }]);
+      let res = new Dict();
+
+      let colorSpace = ColorSpace.parse(cs, xref, res);
+
+      let testSrc = new Uint8Array([27, 125, 250,
```

Comment:
*This applies to other places in the patch as well.*

Put the part from 27 onward on the next line.